src/controllers/cart.js

```diff
@@ -78,37 +76,35 @@ exports.putCartItemByDishId = async (req, res, next) => {
   }
 };
 
-exports.getCartItemByDishId = async (req, res) => {
+exports.getCartItemByDishId = async (req, res, next) => {
   try {
     const cart = await Cart.findOne({
       customerId: req.user.id,
-      'cartItems.dishId': req.params.id,
+      'cartItems.dishId': req.body.id,
     });
     if (cart.cartItems[0]) {
-      res.status(200).json(cart.cartItems[0]);
-    } else {
-      res.status(404).json({ message: 'item not found' });
+      return res.status(200).json(cart.cartItems[0]);
     }
+    return res.status(404).json({ message: 'item not found' });
   } catch (err) {
-    res.status(500).json({ message: 'Failed to get item', error: err });
+    return next(new CustomError('Failed to get item'));
```

Comment:
same thing here instead of 'Failed to delete cart' use err.message and 500 status code